modules/services/window-managers/leftwm.nix

```diff
@@ -0,0 +1,165 @@
+        '';
+        example = literalExpression ''
+          {
+            margin = 5;
+            border_width = 10;
+            default_border_color = "#37474F";
+            floating_border_color = "#225588";
+            focused_border_color = "#885522";
+          }
+        '';
+      };
+    };
+  };
+in {
+  meta.maintainers = [ maintainers.autumnal ];
+
+  options = {
+    xsession.windowManager.leftwm = {
+      enable = mkEnableOption "leftwm window manager";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.leftwm;
+        defaultText = literalExpression "pkgs.leftwm";
+        description = ''
+          LeftWM package to use.
+          </para>
+          <para>
+          The <link xlink:href="https://github.com/leftwm/leftwm/">LeftWM GitHub Repo</link> is a flake and can be used directly.
+        '';
+      };
+
+      settings = mkOption {
+        type = tomlFormat.type;
+        default = { };
+        example = literalExpression ''
+          {
+            modkey = "Mod4";
+            keybind = [
+              {
+                command = "Execute";
+                value = "${pkgs.alacritty}/bin/alacritty";
+                modifier = ["modkey" "Shift"];
+                key = "Return";
+              }
+            ];
+          }
+        '';
+        description = ''
+          LeftWM settings.
+          </para>
+          <para>
+          See <link xlink:href="https://github.com/leftwm/leftwm/wiki/Config"/>.
+          Nix config is almost equivalent to "short syntax" shown in <link xlink:href="https://github.com/leftwm/leftwm/wiki/Config">LeftWM Wiki</link>.
+        '';
+      };
+
+      themes = mkOption {
+        type = types.attrsOf themeSubmodule;
+        default = { };
+        description = ''
+          Theme configuration.
+          The keys of the attributes are the name of the theme.
+          </para>
+          <para>
+          <varname>"current"</varname> is the default theme used by LeftWM. 
+          Symlinking themes to <code>$HOME/.config/leftwm/themes/current</code> instead is recommended by LeftWM.
+        '';
+        example = literalExpression ''
+          {
+            "current" = { ... }
+            "nord" = { ... }
+            "onehalf" = { ... }
+          }
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (hm.assertions.assertPlatform "xsession.windowManager.leftwm" pkgs
+        platforms.linux)
+    ];
+
+    xdg.configFile = mkMerge ([
+      # LeftWM Config
+      ({
+        "leftwm/config.toml" = mkIf (cfg.settings != { }) {
+          source = tomlFormat.generate "leftwm-config" cfg.settings;
+        };
+      })
+      # Themes
+      (listToAttrs (flatten (mapAttrsToList (name: theme: [
+        {
+          name = "leftwm/themes/${name}/up";
+          value = {
+            text = theme.up;
+            executable = true;
+          };
+        }
```

Comment:
If the type of the `up` option is changed as described in another comment, then this will need to be changed to

``` nix
{
  name = "...";
  value = { executable = true; } //
    (if isDerivation theme || isStorePath theme then
      { source = theme; }
    else
      { text = theme; });
}
```

Feel free to put this in a function / format / fix-up the logic.